### PR TITLE
Scaffold of code conventions documentation 

### LINKED ISSUE #8: Coding standards / Style guide
- [ ] Document auto formatting tools and rules sets per language
- [ ] Document team best practices (ie. region comments, alphabetical within regions, etc...)

### PR DESCRIPTION
**Pull request by wintondeshong**
Scaffold of code conventions documentation and some baseline examples of configuration 

Fixes #8 

- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [-] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested (markdown)
- [-] Automated tests are passing
- [-] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [-] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

**Comment by wintondeshong** [comment]
@HeyKos @Stefanie899 @brandongregoryscott - I've gone ahead and put our discussion in issue #8 as well as learnings from projects into the form of code-convention documentation. This is in no way considered "complete", but a solid start. Before I go ahead sending this around to all of engineering for review, I'd love it if you three could review it being your involvement thus far.

**Comment by brandongregoryscott** [comment]
This is a great start @wintondeshong 👍 Glad you pulled in the configurations, git hook, etc. so they're easily available for other projects to reference/copy over.

File: .vscode/typescript.code-snippets [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
```
Comment (HeyKos) [inline .vscode/typescript.code-snippets:1]: I love the inclusion of snippets! Thanks for documenting how to add additional ones.

File: CODE-CONVENTIONS-CSHARP.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+```csharp
+#region Private Properties
+
+// ...
+
+#endregion Private Properties
+```
+
+1. Private properties
```
Comment (HeyKos) [inline CODE-CONVENTIONS-CSHARP.md:19]: There are some interesting concepts in Clean Code that I think we should discuss that could change the structure of the components that make up a class. Obviously nothing needs to change now, but I do think we should discuss at an upcoming TA team meeting.

File: CODE-CONVENTIONS.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
+# Code Conventions & Best Practices
+
+> The purpose of this and related documents is to outline the established programming conventions for the various languages used by andculture engineering.
+
+---
+
+## General
+
+Our approach to any language used at andculture should be to use the respective community as a guide. By using community established conventions and styles, documentation, onboarding, tooling, etc... just fall into place.
```
Comment (HeyKos) [inline CODE-CONVENTIONS.md:9]: I think this is a great way to phrase our approach. Thanks for putting this together.

File: CODE-CONVENTIONS.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+### Formatting
+
+The goal is to maintain auto-formatting no matter what. So if a decision results in needing to disable auto-formatting, it should be avoided.
+
+We try our best to leverage prettier to handle auto-formatting. Check out the [.prettierrc](.prettierrc) file as it contains our common configuration.
+
+Using a [git pre-commit hook](.githooks/pre-commit) we ensure prettier is run on the entire source tree before being commited into source control.
```
Comment (HeyKos) [inline CODE-CONVENTIONS.md:17]: This is awesome!

File: CODE-CONVENTIONS.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+### Organization
+
+When organizing a given file, we try to use these general rules of thumb
+
+#### Alphabetization
```
Comment (HeyKos) [inline CODE-CONVENTIONS.md:23]: This is related to the comment I made above about some of the advice in Clean Code. The author makes the point that functions/methods should be ordered based on the caller/callee relationship to enhance readability. Not saying we need to change course, but worth a discussion.

**Pull request merged**